src/main/java/org/truffleruby/core/numeric/BignumNodes.java

```diff
@@ -700,20 +701,37 @@ public Object compareFallback(DynamicObject a, DynamicObject b) {
     @Primitive(name = "bignum_pow")
     public static abstract class BignumPowPrimitiveNode extends PrimitiveArrayArgumentsNode {
 
-        private final ConditionProfile negativeProfile = ConditionProfile.createBinaryProfile();
+        private static final int BIGLEN_LIMIT = 32 * 1024 * 1024;
```

Comment:
What's this magic number?

Reply:
It's used by MRI. I'm trying to adhere to their semantics for overflow. I'll add a comment.